src/StreamJsonRpc/Protocol/JsonRpcMessage.cs

```diff
@@ -14,6 +16,11 @@ namespace StreamJsonRpc.Protocol;
 [KnownType(typeof(JsonRpcRequest))]
 [KnownType(typeof(JsonRpcResult))]
 [KnownType(typeof(JsonRpcError))]
+#pragma warning disable CS0618 //'KnownSubTypeAttribute.KnownSubTypeAttribute(Type)' is obsolete: 'Use the generic version of this attribute instead.'
```

Comment:
I was afraid that multitargeting libraries would suffer from this. I wonder if I should remove the `Obsolete` attribute and use a custom analyzer instead, with a custom code that you can suppress in a multitargeting library like this.
https://github.com/AArnott/Nerdbank.MessagePack/issues/185

Reply:
I wasn't sure if I should #ifdef based on framework, or just use the netstandard2.0 API and silence the obsolete warning.

Reply:
Using `#if` will help ensure your code is more correct since .NET 8+ APIs are more constrained and will report errors at compile-time that the ns2.0 APIs will not report until runtime. But the choice is yours.